account_payment_order/models/account_payment_order.py

```diff
@@ -162,10 +146,18 @@ def create(self, vals):
     @api.onchange('payment_mode_id')
     def payment_mode_id_change(self):
         journal_id = False
+        res = {'domain': {
+            'journal_id': "[('id', '=', False)]",
```

Comment:
If I make your modification, then I have access to all journals when no payment mode is selected, which is not what I want : I want the user to select a payement mode first. So I prefer my solution, which works well.